woocommerce/woocommerce-android
# Remove eventbus and stateful-ness for Update Variation action

Partially implements  https://github.com/woocommerce/woocommerce-android/issues/6064

Remove the "do not merge" label after merging [the FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2319) and updating the hashcode in build.gradle.

This PR refactors updateVariation into a suspendable function - removes usage of the event bus.

There are two reasons for this change

1. [The event bus architecture is considered as legacy/deprecated](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture)
2. WCAndroid app uses stateful repositories which need to be registered to the event bus - this leads to memory leaks when the repository doesn't unregister from the event bus. It also causes side-effects when multiple instances of a repository are created - eg. duplicate tracking events.


## How to test
1. Go to Product list
2. Open a Variation Product
3. Tap on Variations row
4. Tap on one of the variations
5. Modify something - eg. description
6. Tap on Update
7. Verify "Variation updated" snackbar is shown and the change is visible in WPAdmin
